sdk/src/ctx.rs

```diff
@@ -0,0 +1,51 @@
+//! Context and Context Extensions are a set of traits and utilities that provide a common interface for interacting with the Gadget SDK.
+//!
+//! The Context is the primary interface for interacting with the Gadget SDK and provides access to the various modules and utilities that are available.
+//! Usually, when you need access to the SDK, you will need to pass the Context to your jobs/functions. In your code, you will create a struct that encapsulates all the things that you would need from outside world from your job.
+//! for example, if you need to interact with the network, you will need to have a network client in your struct. If you need to interact with the database storage, you will need to have a db client in your struct. And so on.
+//!
+//! This module provides a set of traits that you can implement for your struct to make it a context-aware struct by adding new functionalities to it.
```

Comment:
This could be simplified to just say it allows you to create custom context types that can be used throughout the SDK.

Reply:
Will improve the docs in the next PRs